src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs

```diff
@@ -120,10 +188,18 @@ public string GetDotNetFrameworkName(IFrameworkNameProvider mappings)
 
                 parts.Add(string.Format(CultureInfo.InvariantCulture, "Version=v{0}", GetDisplayVersion(framework.Version)));
 
-                if (!string.IsNullOrEmpty(framework.Profile))
+                if (!IsNet5Era && !string.IsNullOrEmpty(framework.Profile))
                 {
                     parts.Add(string.Format(CultureInfo.InvariantCulture, "Profile={0}", framework.Profile));
                 }
+                if (IsNet5Era && !string.IsNullOrEmpty(framework.Platform))
```

Comment:
Similarly, I'm not sure there's much value in restricting platform to .NET 5 era only. 
Don't feel strongly about this (yet :P ), but mostly curious if the added perspective makes you reconsider this.

Reply:
Especially given that this is immutable and there's absolutely no way to create a framework object with both a platform and profile at this point.